model/riscv_insts_aext.sail

```diff
@@ -70,26 +70,12 @@ function clause execute(LOADRES(aq, rl, rs1, width, rd)) = {
   // This is checked during decoding.
   assert(width_bytes <= xlen_bytes);
 
-  /* Get the address, X(rs1) (no offset).
-    * Extensions might perform additional checks on address validity.
-    */
-  match ext_data_get_addr(rs1, zeros(), Read(Data), width_bytes) {
-    Ext_DataAddr_Error(e)  => Ext_DataAddr_Check_Failure(e),
-    Ext_DataAddr_OK(vaddr) => {
-      /* "LR faults like a normal load, even though it's in the AMO major opcode space."
-        * - Andrew Waterman, isa-dev, 10 Jul 2018.
-        */
-      if not(is_aligned(bits_of(vaddr), width))
-      then Memory_Exception(vaddr, E_Load_Addr_Align())
-      else match translateAddr(vaddr, Read(Data)) {
-        TR_Failure(e, _)    => Memory_Exception(vaddr, e),
-        TR_Address(addr, _) =>
-          match mem_read(Read(Data), addr, width_bytes, aq, aq & rl, true) {
-            Ok(result) => { load_reservation(bits_of(addr)); X(rd) = sign_extend(result); RETIRE_SUCCESS },
-            Err(e)     => Memory_Exception(vaddr, e),
-          },
-      }
-    }
+  match vmem_read(rs1, zeros(), width_bytes, aq, aq & rl, true) {
```

Comment:
Isn't this missing `Read(Data)`?

Reply:
Since `vmem_read` is supposed to be a high-level utility API, it doesn't require the caller to specify `Read(Data)` (and similarly for `vmem_write`).

Reply:
I think the access type *is* part of the high level API though. For example we use a different value for loading/storing capabilities in CHERI.

Ultimately I think we will end up with `AccessType` something like this:

```
union AccessType ('a : Type) = {
  Read            : 'a,
  Write           : 'a,
  // (Read, Write, operation)
  AMO           : ('a, 'a, amoop),
  // (acquire, release)
  LoadReserved : ('a, bool, bool),
  StoreConditional : ('a, bool, bool),
  InstructionFetch : unit,
  Cache         : CacheAccessType,
  JumpTableFetch : 'a,
}
```

and then remove the `aq`, `rel`, etc. flags. We don't have to do that now but I think we should have the parameter at least so it is easier to fix later.

Reply:
Yeah, we can address those issues later.  The API may also need tweaking for pointer masking, etc.  Let's get the simple version in first.

Reply:
I think it's easy enough to add back now - I can do it if you like? It's just lifting the parameter.

Reply:
Added.